app/src/main/java/org/thoughtcrime/securesms/onboarding/messagenotifications/MessageNotificationsActivity.kt

```diff
@@ -12,22 +12,28 @@ import org.session.libsession.utilities.TextSecurePreferences
 import org.thoughtcrime.securesms.ApplicationContext
 import org.thoughtcrime.securesms.BaseActionBarActivity
 import org.thoughtcrime.securesms.home.HomeActivity
+import org.thoughtcrime.securesms.home.startHomeActivity
 import org.thoughtcrime.securesms.notifications.PushRegistry
+import org.thoughtcrime.securesms.onboarding.loading.LoadingActivity
+import org.thoughtcrime.securesms.onboarding.loading.LoadingManager
 import org.thoughtcrime.securesms.ui.setComposeContent
 import org.thoughtcrime.securesms.util.setUpActionBarSessionLogo
+import org.thoughtcrime.securesms.util.start
 import javax.inject.Inject
 
 @AndroidEntryPoint
 class MessageNotificationsActivity : BaseActionBarActivity() {
 
     @Inject lateinit var pushRegistry: PushRegistry
+    @Inject lateinit var prefs: TextSecurePreferences
+    @Inject lateinit var loadingManager: LoadingManager
 
     private val viewModel: MessageNotificationsViewModel by viewModels()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setUpActionBarSessionLogo(true)
-        TextSecurePreferences.setHasSeenWelcomeScreen(this, true)
+        prefs.setHasSeenWelcomeScreen(true)
```

Comment:
What is the welcome screen?

Reply:
I don't know exactly, this Notifications activity apparently.

Reply:
If we have changed the flow of screens and the logic for the order of screens we should be careful to keep things aligned else we will lose context.
What is this flag used for? Should it be renamed if it is indeed intended to mark the notifications style screen?

Reply:
@bemusementpark

Reply:
IDK, it needs to be set for some reason, or I think you get kicked out of any `SignedInActivity`, seems kinda pointless, because its usually checked `&&`ing with `hasLocalNumber`.

logic didn't change here.

This screen is still visited in onboarding.